`hp_vendor` doesn't compile for Linux 6.2

```bash
❯ make -C "/lib/modules/6.2.2-arch1-1/build" M="~/Source/hp-vendor/dkms" modules
  CC [M]  ~/Source/hp-vendor/dkms/hp_vendor.o
~/Source/hp-vendor/dkms/hp_vendor.c:92:24: error: initialization of ‘int (*)(struct power_supply *, struct acpi_battery_hook *)’ from incompatible pointer type ‘int (*)(struct power_supply *)’ [-Werror=incompatible-pointer-types]
   92 |         .add_battery = hp_vendor_battery_add,
      |                        ^~~~~~~~~~~~~~~~~~~~~
~/Source/hp-vendor/dkms/hp_vendor.c:92:24: note: (near initialization for ‘hp_vendor_battery_hook.add_battery’)
~/Source/hp-vendor/dkms/hp_vendor.c:93:27: error: initialization of ‘int (*)(struct power_supply *, struct acpi_battery_hook *)’ from incompatible pointer type ‘int (*)(struct power_supply *)’ [-Werror=incompatible-pointer-types]
   93 |         .remove_battery = hp_vendor_battery_remove,
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
~/Source/hp-vendor/dkms/hp_vendor.c:93:27: note: (near initialization for ‘hp_vendor_battery_hook.remove_battery’)
cc1: some warnings being treated as errors
make[1]: *** [scripts/Makefile.build:252: ~/Source/hp-vendor/dkms/hp_vendor.o] Error 1
make: *** [Makefile:2021: ~/Source/hp-vendor/dkms] Error 2
```

Looks the kernel API expects these functions to take an additional argument. 

Compiles fine for 6.1.x.

--- a/dkms/hp_vendor.c
+++ b/dkms/hp_vendor.c
@@ -70,7 +70,11 @@ static struct attribute *hp_vendor_battery_attrs[] = {
 
 ATTRIBUTE_GROUPS(hp_vendor_battery);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
+static int hp_vendor_battery_add(struct power_supply *battery, struct acpi_battery_hook *hook)
+#else
 static int hp_vendor_battery_add(struct power_supply *battery)
+#endif
 {
 	// HP vendor only supports 1 battery
 	if (strcmp(battery->desc->name, "BATT") != 0)
@@ -82,7 +86,11 @@ static int hp_vendor_battery_add(struct power_supply *battery)
 	return 0;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
+static int hp_vendor_battery_remove(struct power_supply *battery, struct acpi_battery_hook *hook)
+#else
 static int hp_vendor_battery_remove(struct power_supply *battery)
+#endif
 {
 	device_remove_groups(&battery->dev, hp_vendor_battery_groups);
 	return 0;

